go/cli/mcap/cmd/info_test.go

```diff
@@ -0,0 +1,94 @@
+package cmd
+
+import (
+	"bytes"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/foxglove/mcap/go/mcap"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestInfo(t *testing.T) {
+	cases := []struct {
+		assertion string
+		inputfile string
+		expected  string
+	}{
+		{
+			"OneMessage",
+			"../../../../tests/conformance/data/OneMessage/OneMessage-ch-chx-mx-pad-rch-rsh-st-sum.mcap",
+			`library:
+profile:
+messages:  1
+duration:  0s
+start:     0.000000002
+end:       0.000000002
+compression:
+	: [1/1 chunks] [115.00 B/115.00 B (0.00%)]
+channels:
+	(1) example  1 msgs (+Inf Hz)   : Example [c]
+attachments: 0
+metadata: 0`,
+		},
+		{
+			"OneSchemalessMessage",
+			"../../../../tests/conformance/data/OneSchemalessMessage/OneSchemalessMessage-ch-chx-mx-pad-rch-st.mcap",
+			`library:
+profile:
+messages:  1
+duration:  0s
+start:     0.000000002
+end:       0.000000002
+compression:
+	: [1/1 chunks] [70.00 B/70.00 B (0.00%)]
+channels:
+	(1) example  1 msgs (+Inf Hz)   : <no schema>
+attachments: 0
+metadata: 0`,
+		},
+		{
+			"OneSchemalessMessage_NoChannels",
+			"../../../../tests/conformance/data/OneSchemalessMessage/OneSchemalessMessage.mcap",
+			`library:
+profile:
+channels:
+attachments: unknown
+metadata: unknown`,
```

Comment:
I'm a bit nervous this will become annoying to update as the info command gains new functionality. This text display isn't really stable - for that we should probably introduce some kind of specification for the output.

Would it be sufficient to test that printInfo simply doesn't crash the program? If you're interested in asserting on the name of the spoofed schema, you should be able to do that by making an assertion against the info structure's Schemas field.

Reply:
I wonder whether if we wrote that test to loop over all of the conformance files, we'd find more bugs or if this is the only one. That seems like it would be a stronger test to me, but I don't want to obligate you to address whatever else it might show.

Reply:
I think both of these are good points!
I made the test run on every `.mcap` in the conformance folder and only found one new crash, so I fixed it